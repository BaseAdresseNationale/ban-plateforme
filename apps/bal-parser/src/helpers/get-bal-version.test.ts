import fs from 'node:fs';
import { describe, expect, test } from 'vitest';
import { getBalVersion } from './index.js';

const pathToMockBalJSON1 = './apps/bal-parser/data-mock/adresses-21286_cocorico.json';
const mockBalJSONstr1 = fs.readFileSync(pathToMockBalJSON1, 'utf8');
const balJSON1 = JSON.parse(mockBalJSONstr1);

const pathToMockBalJSON2 = './apps/bal-parser/data-mock/adresses-21286_cocorico.1.4.json';
const mockBalJSONstr2 = fs.readFileSync(pathToMockBalJSON2, 'utf8');
const balJSON2 = JSON.parse(mockBalJSONstr2);

describe('balTopoToBanTopo', () => {
  test('Should return version 1.3', async () => {
    expect(getBalVersion(balJSON1)).toMatchSnapshot();
  });

  test('Should return version 1.4', async () => {
    expect(getBalVersion(balJSON2)).toMatchSnapshot();
  });
});
