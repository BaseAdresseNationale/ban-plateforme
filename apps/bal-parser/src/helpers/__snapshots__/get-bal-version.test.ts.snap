// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`balTopoToBanTopo > Should return version 1.3 1`] = `"1.3"`;

exports[`balTopoToBanTopo > Should return version 1.4 1`] = `"1.4"`;
